### PR TITLE
Release 2.0

### PR DESCRIPTION
**Pull request by ericpre**
https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/releasing_guide.md

I merge the changes for 2.0rc0 and 2.0 together as it makes the release notes slighlty scattered and the difference between 2.0rc0 and 2.0 is only relevant for developers.

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -193,7 +193,7 @@
         "version_match": version,
     },
     "navbar_start": ["navbar-logo", "version-switcher"],
-    "announcement": "HyperSpy API is changing in version 2.0, see <a href='https://hyperspy.org/hyperspy-doc/dev/changes.html'>upcoming changes!</a>",
+    "announcement": "HyperSpy API is changing in version 2.0, see the <a href='https://hyperspy.org/hyperspy-doc/current/changes.html'>release notes!</a>",
```
Comment (ericpre) [inline doc/conf.py:196]: This will work when the documentation is pushed to https://github.com/hyperspy/hyperspy-doc.

**Comment by jlaehne** [comment]
Bump version of `setup-miniconda` in `tests_extension.yml` (#3277) and `setup-python` in `tests.yml` (#3279)?
All other bumped dependencies have moved to the reusable workflows.

**Comment by ericpre** [comment]
I was thinking to do this after the release, because these PRs are based on another branch.

**Comment by jlaehne** [comment]
> I was thinking to do this after the release, because these PRs are based on another branch.

Thought it could be tackled along the way, but should not be so urgent and can as well be sorted out after updating RnMinor (most of these PRs are obsolete then anyway, due to the reusable workflows).

**Comment by ericpre** [comment]
https://pypi.org/project/hyperspy/2.0

**Pull request merged**